tests/test_eradiate/regression/romc/test_het01.py

```diff
@@ -0,0 +1,88 @@
+import pathlib
+import pytest
+import numpy as np
+import os
+import xarray as xr
+
+from eradiate.test_tools.regression import Chi2Test
+from eradiate.data import data_store
+from eradiate.experiments import RamiExperiment
+from eradiate.units import unit_registry as ureg
+
+
+@pytest.mark.regression
+def test_het01_brfpp(mode_mono_double, metadata, session_timestamp):
+
+    leaf_spec_path = data_store.fetch("tests/canopies/HET01/HET01_UNI_sphere.def")
+    leaf_pos_path = data_store.fetch("tests/canopies/HET01/HET01_UNI_instances.def")
+
+    exp = RamiExperiment(
+        canopy={
+            "type": "discrete_canopy",
+            "instanced_canopy_elements": [
+                {
+                    "construct": "from_file",
+                    "filename": leaf_pos_path,
+                    "canopy_element": {
+                        "type": "leaf_cloud",
+                        "construct": "from_file",
+                        "filename": leaf_spec_path,
+                        "leaf_reflectance": 0.4957,
+                        "leaf_transmittance": 0.4409,
+                        "id": "spherical_leaf_cloud",
+                    },
+                },
+            ],
+            "size": [100, 100, 30] * ureg.m,
+        },
+        surface={"type": "lambertian", "reflectance": 0.159},
+        padding=20,
+        measures=[
+            {
+                "type": "mdistant",
+                "construct": "from_viewing_angles",
+                "spp": 10000,
+                "azimuths": [180] * ureg.deg,
+                "zeniths": np.arange(-75, 75.01, 2) * ureg.deg,
+                "target": {
+                    "type": "rectangle",
+                    "xmin": -50 * ureg.m,
+                    "xmax": 50 * ureg.m,
+                    "ymin": -50 * ureg.m,
+                    "ymax": 50 * ureg.m,
+                    "z": 30 * ureg.m,
+                },
+            }
+        ],
+        illumination={
+            "type": "directional",
+            "zenith": 20 * ureg.deg,
+            "azimuth": 0.0 * ureg.deg,
+            "irradiance": 20.0,
+        },
+        integrator={"type": "path", "max_depth": -1},
+    )
+
+    exp.run()
+    result = exp.results["measure"]
+
+    reference_path = data_store.fetch("tests/canopies/HET01/het01_brfpp_ref.nc")
+    reference = xr.load_dataset(reference_path)
+    archive_path = metadata.get("archive_path", None)
+
+    if archive_path:
+        datestring = session_timestamp.strftime(format="%Y%m%d")
+        timestring = session_timestamp.strftime(format="%H%M%S")
+        archive_filename = f"{datestring}-{timestring}-het01.nc"
+        archive_filename = os.path.join(archive_path, archive_filename)
+    else:
+        archive_filename = None
```

Comment:
```suggestion
archive_filename = os.path.join(
    archive_path, f"{session_timestamp:%Y%m%d-%H%M%S}-het01.nc"
) if archive_path else None
```
(to be confirmed)

See also

- https://www.python.org/dev/peps/pep-3101/#controlling-formatting-on-a-per-type-basis
- https://docs.python.org/3/library/datetime.html#datetime.date.__format__